### PR TITLE
Making the MIM test recipe for Security Baseline module test in full ASB v2, including confirming audits passing

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

Making the module tester to be able to run the test recipe for Security Baseline and validate results in full for audit, except the 4 special case checks missing automatic remediation, alongside existing remediation verification.

Also fixing couple of memory leaks in the test and improving test logging for ASB.

Example of last line for an audit reported MIM object check pass:

[2024-06-13 14:26:19] [main.c:467] Assertion passed with reason: 'None of the requested user accounts ('games,test') is present'

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/test/main.c [hunk 1]
```diff
@@ -418,21 +431,53 @@ int RunTestStep(const TEST_STEP* test, const MANAGEMENT_MODULE* module)
...
+                    LOG_INFO("Assertion passed with reason: '%s'", reason + strlen(SECURITY_AUDIT_PASS));
+                }
+            }
+            else if (actualJsonValue != NULL)
             {
                 if (NULL == (expectedJsonValue = json_parse_string(test->payload)))
                 {
                     LOG_ERROR("Failed to parse expected JSON payload: %s", test->payload);
-                    result = EINVAL;
+                    result = -1;
```
Comment (AhmedBM) [inline src/modules/test/main.c:475]: `EINVAL` seems like a better choice here? Why change?

Comment (MariusNi) [reply src/modules/test/main.c:475]: For consistency. This test uses -1 for test pass failures and in the end does not matter. If rest of this test code would use real errno errors, sure. As it is this looked like a sore thumb.

Comment (MariusNi) [reply src/modules/test/main.c]: Replaced all -1 returns in this function with EFAULT

**Review by robertschaedler3 — APPROVED** [review]

**Review by AhmedBM — APPROVED** [review]
👌

**Pull request merged**